hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -761,10 +736,16 @@ public long heapSize() {
   }
 
   /**
-   * @return true to indicate the block is allocated from JVM heap, otherwise from off-heap.
+   * Will be override by {@link SharedMemHFileBlock} or {@link ExclusiveMemHFileBlock}. Return true
+   * by default.
    */
-  boolean isOnHeap() {
-    return buf.hasArray();
+  public boolean isSharedMem() {
```

Comment:
Still we create HFileBlock only in tests? If not we could have left this API abstract only and even HFileBlock.  That would have been best. May be in a follow up we can do that. Just renaming will be enough. In this jira keep it this way only. Fine.